src/librustdoc/html/render.rs

```diff
@@ -431,18 +435,20 @@ struct Type {
     generics: Option<Vec<String>>,
 }
 
-impl ToJson for Type {
-    fn to_json(&self) -> Json {
-        match self.name {
-            Some(ref name) => {
-                let mut data = Vec::with_capacity(2);
-                data.push(name.to_json());
-                if let Some(ref generics) = self.generics {
-                    data.push(generics.to_json());
-                }
-                Json::Array(data)
+impl Serialize for Type {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        if let Some(name) = &self.name {
+            let mut seq = serializer.serialize_seq(None)?;
+            seq.serialize_element(&name)?;
+            if let Some(generics) = &self.generics {
+                seq.serialize_element(&generics)?;
```

Comment:
Hmm, could this serialize either `(name,)` or `(name, generics)` depending on whether the latter is present? Although this might look a bit silly.

Reply:
Oh, also, doesn't `serde` have ways of configuring how `enum` variants are serialized, so that you can just use `#[derive(Serialize)]` everywhere?